server/core/src/main/scala/org/infinispan/server/core/UnsignedNumeric.java

```diff
@@ -0,0 +1,91 @@
+package org.infinispan.server.core;
+
+import io.netty.buffer.ByteBuf;
+
+import java.util.OptionalInt;
+import java.util.OptionalLong;
+
+/**
+ * // TODO: Document this
+ *
+ * @author wburns
+ * @since 4.0
+ */
+public class UnsignedNumeric {
+   /**
+    * Reads an int stored in variable-length format.  Reads between one and five bytes.  Smaller values take fewer
+    * bytes.  Negative numbers are not supported.
+    */
+   public static int readUnsignedInt(ByteBuf in) {
+      byte b = in.readByte();
+      int i = b & 0x7F;
+      for (int shift = 7; (b & 0x80) != 0; shift += 7) {
+         b = in.readByte();
+         i |= (b & 0x7FL) << shift;
+      }
+      return i;
+   }
+
+   /**
+    * Same as {@link #readUnsignedInt(ByteBuf)} except that it returns an <code>OptionalInt</code> that can be used
+    * when the byte buf is not used in blocking mode.  If an empty value is returned the reader index will be reset.
+    * @param in the byte buf to read from
+    * @return an optional showing whether or not the int could be read
+    */
+   public static OptionalInt readOptionalUnsignedInt(ByteBuf in) {
```

Comment:
I added it to mirror the other UnsignedNumeric in core since we had both Int and Long.  I figured it is easiest to add it now when I was actually thinking about it rather than later, and it can be used by others if needed.